src/utils.ts

```diff
@@ -100,6 +100,10 @@ export const isArrayForMap = (value: unknown): value is [any, any][] => {
   return false;
 };
 
+export const isArrayForSet = (value: unknown): value is any[] => {
+  return Array.isArray(value);
```

Comment:
Would this be better 🤷‍♂️ 

```suggestion
  return Array.isArray(value) && value.every(item => !Array.isArray(item));

Reply:
@codeBelt Hi, it seems like it's possible to create a set of arrays, even though I can't think of a use case where it's needed.

<img width="365" alt="Screenshot 2025-03-11 at 14 09 52" src="https://github.com/user-attachments/assets/f54a4bfe-2973-4008-97ab-812499b748eb" />

The function `isArrayForSet` isn't used to determine whether a variable is an actual Set. It's part of the `instanceof ObservableSet` check. So, it doesn't seem to me that extra checks are necessary here

```js
if (target[property.key] instanceof ObservableSet && isArrayForSet(propertyData))
```